### ISSUE TITLE
ema_smooth uses .T instead of moveaxis, corrupts rank-3+ inputs

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#155]
## What's wrong

\`alberta_framework/streams/gauntlet.py::ema_smooth\` is documented as "Exponential-moving-average smoothing along the last axis" and is meant to work for arbitrary-rank arrays. It moves the scan axis to the front by transposing:

\`\`\`python
_, smoothed = jax.lax.scan(step, values[..., 0], values.T)
return smoothed.T
\`\`\`

\`.T\` reverses *all* axes, not just the last one. For a rank-2 input this happens to be identical to moving only the last axis to the front (\`moveaxis(values, -1, 0)\`), which is exactly why none of the current 1D/2D callers caught it. For a rank-3+ input, \`.T\` scrambles the batch structure instead of relocating only the scan axis.

Confirmed independently:

\`\`\`text
>>> values.shape
(2, 3, 4)
>>> values.T.shape
(4, 3, 2)          # wrong: batch and feature axes swapped
>>> jnp.moveaxis(values, -1, 0).shape
(4, 2, 3)          # correct: only the last axis moved to front
```

For a rank-2 \`(2, 3)\` input both approaches give an identical \`(3, 2)\`, so no existing caller has ever observed the divergence.

## Reachability

\`ema_smooth\` is called by \`steps_to_criterion\`, which is called by the publicly-exported \`gauntlet_scorecard\` (re-exported from \`alberta_framework/streams/__init__.py\` along with \`run_gauntlet\`/\`run_gauntlet_batched\`). Honesty check: every current real call path into \`steps_to_criterion\` passes rank-2 \`sq_errors\` (shape \`(n_seeds, num_steps)\` per \`run_gauntlet_batched\`'s own docstring), so this bug is not currently producing wrong numbers in the shipped gauntlet pipeline today. It's a latent correctness bug in a publicly-exported, general-purpose utility function whose docstring promises "along the last axis" for any rank — the first rank-3+ caller (batched-over-an-extra-axis usage, which the function's own signature invites) will silently get corrupted output with no error.

## Fix

Use \`jnp.moveaxis(values, -1, 0)\` / \`jnp.moveaxis(smoothed, 0, -1)\` instead of \`.T\`, which only ever moves the target axis and is correct for any rank.

PR incoming with a rank-3 regression test and a mutation-resistance transcript.
